### DISCUSSION TITLE
High loss rate

### DISCUSSION THREAD

**Discussion by TorbjornHouge** [mavlink-router/mavlink-router#407] [General]
Hi. I am using PX4 on a Pixhawk Orange Cube, Raspberry Pi 4 running mavlink-router, and for testing I connect it to my laptop using an Ethernet cable (I get the problem with TCP and UDP). I am experiencing high loss rates, and more often than not the parameter download to QGC fails. I have no idea what may be wrong at this point, but my setup og mavlink router is one of the options.

Is this familiar to anyone and does anyone have suggestions on how to solve it?

**Comment by jbeyerstedt** [comment]
Please post the mavlink-router config, otherwise we can only guess. And do you use another connection to PX4 like a telemetry radio while you're plugged in via ethernet?

**Comment by TorbjornHouge** [comment]
Thank you for the quick reply. I didn't have the system on hand so I had to copy off the config now. Mavlink-router is running as a service.

When plugged in via ethernet, I don't use a telemetry radio. In the field, we use Ubiquiti IP radios instead of wired ethernet, but since the problem is the same I choose to fault-find using ethernet to avoid the chance of a bad link to be the cause. We have used a secondary telemetry link before, so TELEM2 is configured to output telemetry, but nothing is connected. Eventually we would like to have the redundancy working but it is not important for now.

The config file is (for readability I have removed the commented text before posting it here):

```
[General]

[UartEndpoint alpha]
Device = /dev/ttyAMA0
Baud = 57600,115200

[UdpEndpoint bravo]
Mode = Server
Address = 0.0.0.0
Port = 14550

[UdpEndpoint charlie]
Mode = Normal
Address = 127.0.0.1
Port = 14551
```

When using, I connect to endpoint charlie.
I can am also able to use TCP at port 5760, however this does not give much improvement, if any.

**Reply by jbeyerstedt** [reply]
Config looks fine, but your ground station PC can't connect to endpoint Charlie as this is sending to localhost on the raspberry pi. You could only connect to raspberry pi UDP port 14550 or TCP port 5760.

**Reply by TorbjornHouge** [reply]
Sorry, that is correct. Typo on my side, I connect to endpoint Bravo at 14550.

**Reply by TorbjornHouge** [reply]
QGroundControl reports loss rates of 20 - 30 % (with different PCs, so it is not likely it is the individual PC setup at fault here), which is horrible. It may not be Mavlink Router, but something is not right.

**Reply by jbeyerstedt** [reply]
To rule out mavlink-router you could look at mavlink-router's logs if there are messages to unknown endpoints. If you start mavlink-router in verbose mode you'll see which systems were discovered on which endpoint.

**Reply by TorbjornHouge** [reply]
Thank you. Further investigation rules out mavlink-router by getting the same result connecting directly to the serial port out from the autopilot. Strange. So I will take this directly with the PX4-community onward, but it is very good to be able to rule out mavlink-router.
